keps/sig-storage/20200709-pod-pvc-qos-limits.md

```diff
@@ -0,0 +1,255 @@
+        - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+        - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+    - [Upgrade / Downgrade / Version Skew Strategy](#upgrade--downgrade--version-skew-strategy)
+      - [Interaction with old <code>AttachVolumeLimit</code> implementation](#interaction-with-old--implementation)
+  - [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+## Summary
+
+Better resource management for iops or bps of block device in pvc.
+In some senarios, we need to limit PV's iops limit or Pod iops on a device or volume at runtime.
+
+For container runtime, dockerd provides blkio related params in docker run to limit a device's iops and bps. [docker reference #block-io-bandwidth-blkio-constraint](https://docs.docker.com/engine/reference/run/#block-io-bandwidth-blkio-constraint)
+
+
+## Motivation
+The proposal would benifit in below two senarios.
+
+* Local Storage(local disk devices), better speed:
+For instances, I want to use local storage(local device) to gain the storage local speed.
+These PVCs in my cluster are blocking devices.
+
+* rootfs (preventing being overused by some pods)
+Besides, the rootfs is shared by all pods in your node, the limitation will help you make node more stable. When one pod in your node heavily uses rootfs, it may effect other pods on the same node. The limitation would help as well.
+
+
+### Goals
+
+- The limit is runtime limitation for block device when it is mounted to the pod. 
+- Only blocking device will be limited with specified volume device id. 
+- This should be implemented with cgroup, so it will only work beyond cgroup capability.
+- An extended feature that can be provided later is that we can limit iops of rootfs for container runtime.  This will reduce the influcences between pods， running on the same host.
```

Comment:
- cgroups can limit iops for local block volume and rootfs. 

I may change to goal to this. For all types of volumes' QoS, it may be a much bigger topic.